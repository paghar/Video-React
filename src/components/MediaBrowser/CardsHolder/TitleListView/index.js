import React from 'react';
import {
  ListViewTitle,
  TitleUploader,
  TitleDuration,
  TitleName,
  TitleSize,
  Title,
} from './index.style';
import Typography from 'components/shared/Typography';
import { Generals } from 'theme/colors';
import { Display1 } from 'theme/fonts';
type Props = {
  className?: string,
  clicked?: any => void,
};
const TitleListView = ({ className, clicked }: Props & AssetCardType) => {
  return (
    <ListViewTitle>
      <TitleName width="311px">
        <Typography color={Generals.vii} font={Display1}>
          Name
        </Typography>
      </TitleName>
      <TitleUploader width="112px">
        <Typography color={Generals.vii} font={Display1}>
          Uploader
        </Typography>
      </TitleUploader>
      <Title width="54px">
        <Typography color={Generals.vii} font={Display1}>
          Dialogs
        </Typography>
      </Title>
      <TitleSize width="45px">
        <Typography color={Generals.vii} font={Display1}>
          size
        </Typography>
      </TitleSize>
      <TitleDuration width="57px">
        <Typography color={Generals.vii} font={Display1}>
          Duration
        </Typography>
      </TitleDuration>
      <Title width="86px">
        <Typography color={Generals.vii} font={Display1}>
          UploadDate
        </Typography>
      </Title>
      <Title width="99px">
        <Typography color={Generals.vii} font={Display1}>
          Status
        </Typography>
      </Title>
    </ListViewTitle>
  );
};

export default TitleListView;
