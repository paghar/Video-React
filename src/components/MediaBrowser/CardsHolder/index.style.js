//@flow
import styled from 'styled-components';
import { Generals, Primary } from 'theme/colors';

export const StyledThumb = styled.div`
  border-radius: 10px;
  background-color: ${Generals.vi};
  width: 3px !important;
  opacity: 0.3;
`;
export const StyledTrack = styled.div`
  height: 100%;
  right: 0px;
  background-color: ${Generals.v};
  margin-top: 5px;
  border-radius: 10px;
  width: 2px !important;
  padding: 5px 0px;
`;

export const AssetGutter = styled.hr`
  border: none;
  width: 21px;
  height: 190px;
  content: ' ';
  ${({ isDraggingOver }) =>
      isDraggingOver &&
      `
  :hover::after {
    display: block;
  }
  `}
    ::after {
    content: '';
    position: absolute;
    margin-left: 11px;
    margin-top: 12px;
    width: 2px;
    display: none;
    height: 178px;
    background-color: ${Primary.Dim};
  }
`;
export const FirstAssetGutter = styled(AssetGutter)`
  position: absolute;
  z-index: 999;
  ::after {
    margin-left: 0;
  }
`;
export const CardsWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: ${({ expanded }) => (expanded ? '100%' : '98%')};
  margin-left: ${({ expanded }) => (expanded ? '0.6%' : '30px')};
  margin-right: ${({ expanded }) => (expanded ? '14px' : '0px')};
  margin-bottom: 15px;
  transform: translate3d(0, 0, 0);
  > figure {
    display: flex;
    z-index: 20;
    width: ${({ expanded }) => (expanded ? '97%' : `auto`)};
    height: ${({ expanded }) => (expanded ? '62px' : `auto`)};
    margin-left: ${({ expanded }) => (expanded ? '11px' : '3px')};
    margin-bottom: ${({ expanded }) => (expanded ? '3px' : '6px')};
    margin-right: ${({ expanded }) => (expanded ? '3px' : '0')};
    margin-top: 3px;
    &[data-type='folder'] {
      margin-bottom: ${({ expanded }) => (expanded ? '0' : '9px')};
    }
    &[data-type='asset'] {
      margin-top: ${({ expanded }) => (expanded ? '0' : '6px')};
    }
  }
`;
