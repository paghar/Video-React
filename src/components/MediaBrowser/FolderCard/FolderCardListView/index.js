//@flow
import React from 'react';
import { Display2, Display5 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import { getListViewDate } from 'helpers/time';
import {
  ItemWrapper,
  NameWrapper,
  FileNumberLabel,
  FolderNameLabel,
  FileDialogs,
  StyledIcon,
  MenuToggler,
  FileUploader,
  FileSize,
  FileDuration,
  FileDate,
  FileStatus,
  Duration,
} from './index.style';
type Props = {
  selected?: boolean,
  fileCount: number,
  name: string,
  uploadDate: number,
  uploader: string,
  size?: string,
  className?: string,
  dragging?: boolean,
};
const FolderCardListView = ({
  uploadDate,
  fileCount,
  name,
  className,
  uploader,
  size,
  selected,
  dragging,
}: Props) => {
  return (
    <ItemWrapper dragging={dragging}>
      <NameWrapper>
        <FolderNameLabel
          selected={selected}
          color={Generals.vii}
          font={Display2}
        >
          {name}
        </FolderNameLabel>
        <FileNumberLabel
          selected={selected}
          hoverable
          font={Display2}
          color={Generals.vi}
        >
          {fileCount} files
        </FileNumberLabel>
      </NameWrapper>
      <FileUploader selected={selected} color={Generals.vi} font={Display5}>
        {uploader}
      </FileUploader>
      <FileDialogs>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileDialogs>
      <FileSize selected={selected} color={Generals.vi} font={Display5}>
        {size}
      </FileSize>
      <FileDuration>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileDuration>

      <FileDate selected={selected} color={Generals.vi} font={Display5}>
        {getListViewDate(uploadDate)}
      </FileDate>
      <FileStatus>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileStatus>
      <MenuToggler
        backgroundColor="transparent"
        hoverBackgroundColor="transparent"
        hoverColor={Generals.vii}
        radius="0px"
        iconSize="12px"
        icon="61762"
        spacing="0"
        width="auto"
        height="auto"
        iconColor={Generals.vii}
        hoverIconColor={Generals.vii}
        selected={selected}
      />
    </ItemWrapper>
  );
};

export default FolderCardListView;
