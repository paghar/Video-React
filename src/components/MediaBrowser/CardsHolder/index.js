//@flow
import React, { useRef, useEffect, useState } from 'react';
import Shuffle from 'shufflejs';
import { Scrollbars } from 'react-custom-scrollbars';
import Draggable from 'react-draggable';

import AssetCard from 'components/MediaBrowser/AssetCard';
import FolderCard from 'components/MediaBrowser/FolderCard';
import {
  CardsWrapper,
  StyledThumb,
  StyledTrack,
  AssetGutter,
  FirstAssetGutter,
} from './index.style';
import { findParent } from 'helpers';
import TitleListView from './TitleListView';
import { FOLDER_TYPE } from 'constants/variables';
import {
  selectAssetActionCreator,
  addAssetToSelectionActionCreator,
  startDragActionCreator,
  endDragActionCreator,
} from 'pages/MediaBrowser/index.actions';
import type { AssetCardType } from 'components/MediaBrowser/AssetCard';
import type { FolderCardType } from 'components/MediaBrowser/FolderCard';

export type CardsType = {
  [assetId: string]: (FolderCardType | AssetCardType) & {
    type: string,
    customOrder: number,
  },
};

type Props = {
  expanded: boolean,
  className?: string,
  assetsList: CardsType,
  dispatchAssetsList: any => void,
  dropedOnAsset: (from: any, to: any) => void,
  dropedOnGutter: (from: any, to: any, isFirst: boolean) => void,
};

const CardsHolder = ({
  className,
  assetsList,
  dispatchAssetsList,
  dropedOnAsset,
  dropedOnGutter,
  expanded,
}: Props) => {
  const cardsRef = useRef(); // cards holder element ref
  const shuffleRef: any = useRef(); // shufflejs instance holder
  const [isDragging, setIsDragging] = useState(false);
  const sortByCustomOrder = element => {
    return parseFloat(element.getAttribute('data-order'));
  };
  // create suffle instance on cards
  useEffect(() => {
    shuffleRef.current = new Shuffle(cardsRef.current, {
      useTransforms: false,
      itemSelector: 'figure',
      initialSort: { by: sortByCustomOrder },
    });
    return () => {
      shuffleRef.current.destroy();
    };
  }, []);
  useEffect(() => {
    window.dispatchEvent(new Event('resize'));
  }, [expanded]);
  //style of scroll thumb
  const renderThumb = ({ style }) => {
    return <StyledThumb style={style} />;
  };
  //style of scroll track
  const renderTrackVertical = ({ style }) => {
    return <StyledTrack style={style} />;
  };
  // select an asset by dispatching event to assets store
  const selectAsset = assetId => {
    dispatchAssetsList(selectAssetActionCreator(assetId));
  };
  // dispatch add asset to selection if ctrl key was pressed
  const addAssetToSelection = assetId => {
    dispatchAssetsList(addAssetToSelectionActionCreator(assetId));
  };
  // handle a click on an asset
  const handleAssetClick = (assetId, { ctrlKey, metaKey }) => {
    if (ctrlKey || metaKey) {
      addAssetToSelection(assetId);
    } else {
      selectAsset(assetId);
    }
  };

  // handles on drop element on an other element
  const handleDropZone = (fromAsset, toAsset) => {
    const gutterDroped = findParent(toAsset, 'HR');
    if (gutterDroped) {
      const targetId = findParent(gutterDroped, 'FIGURE').getAttribute(
        'data-id'
      );
      if (targetId)
        dropedOnGutter(
          fromAsset.getAttribute('data-id'),
          targetId,
          gutterDroped.getAttribute('data-id') === '-1'
        );
      shuffleRef.current.sort({ by: sortByCustomOrder });
    } else {
      const figureDroped = findParent(toAsset, 'FIGURE');
      if (figureDroped) {
        const targetId = figureDroped.getAttribute('data-id');
        if (targetId)
          dropedOnAsset(fromAsset.getAttribute('data-id'), targetId);
      }
    }
  };
  // handles when user ends the drag, it reset the style and scalings
  const handleOnDragEnd = ({ target }, { node }) => {
    handleDropZone(node, target);
    const nodeStyle = node.style;
    if (nodeStyle) {
      nodeStyle.transitionDuration = '220ms';
      nodeStyle.transform = 'translate(0,0)';
      nodeStyle.pointerEvents = 'auto';
      nodeStyle.opacity = '1';
      setTimeout(() => (nodeStyle.zIndex = '20'), 500);
      setIsDragging(false);
      dispatchAssetsList(endDragActionCreator(node.getAttribute('data-id')));
    }
    shuffleRef.current.layout();
  };
  // handles on drag start
  const handleOnDragStart = (_, { node }) => {
    node.style.transitionDuration = '0ms';
    startDrag(node);
  };
  // set states and styles on drag start
  const startDrag = node => {
    const nodeStyle = node.style;
    if (nodeStyle) {
      if (!nodeStyle.transitionProperty.includes('transform'))
        nodeStyle.transitionProperty += ', transform, opacity';
      nodeStyle.pointerEvents = 'none';
      nodeStyle.zIndex = '25';
      setIsDragging(node.getAttribute('data-type'));
      dispatchAssetsList(startDragActionCreator(node.getAttribute('data-id')));
    }
  };
  // renders a single folder
  const renderFolder = (assetId, isFirst) => {
    const { customOrder } = assetsList[assetId];
    return (
      <figure
        data-id={assetId}
        data-type="folder"
        data-order={customOrder}
        style={{ order: customOrder }}
      >
        {isFirst && !expanded && (
          <FirstAssetGutter
            isDraggingOver={isDragging}
            data-type="gutter"
            data-id="-1"
          />
        )}
        <FolderCard
          isDraggingOver={isDragging}
          expanded={expanded}
          width={250}
          {...assetsList[assetId]}
          clicked={event => handleAssetClick(assetId, event)}
        />
        {!expanded && (
          <AssetGutter
            isDraggingOver={isDragging}
            data-type="gutter"
            data-id={assetId}
          />
        )}
      </figure>
    );
  };
  // renders a single file
  const renderFile = (assetId, isFirst) => {
    const { customOrder } = assetsList[assetId];
    return (
      <figure
        data-id={assetId}
        data-type="asset"
        style={{ order: customOrder }}
        data-order={customOrder}
      >
        {isFirst && !expanded && (
          <FirstAssetGutter
            isDraggingOver={isDragging}
            data-type="gutter"
            data-id="-1"
          />
        )}
        <AssetCard
          isDraggingOver={isDragging && isDragging !== 'folder'}
          expanded={expanded}
          cardWidth={250}
          {...assetsList[assetId]}
          clicked={event => handleAssetClick(assetId, event)}
        />
        {!expanded && (
          <AssetGutter
            isDraggingOver={isDragging}
            data-type="gutter"
            data-id={assetId}
          />
        )}
      </figure>
    );
  };
  // renders all assets
  const renderAssets = () => {
    const minOrder = Math.min(
      ...Object.keys(assetsList).reduce(
        (prev, current) => [...prev, assetsList[current].customOrder],
        []
      )
    );
    return Object.keys(assetsList).map((assetId, index) => {
      const { customOrder, type } = assetsList[assetId];
      return (
        <Draggable
          position={{ x: 0, y: 0 }}
          onStart={handleOnDragStart}
          onStop={handleOnDragEnd}
          key={assetId}
        >
          {type === FOLDER_TYPE
            ? renderFolder(assetId, customOrder === minOrder)
            : renderFile(assetId, customOrder === minOrder)}
        </Draggable>
      );
    });
  };

  return (
    <>
      {expanded && <TitleListView />}
      <Scrollbars
        style={{ width: '100%' }}
        autoHide
        renderThumbVertical={renderThumb}
        className={className}
        renderTrackVertical={renderTrackVertical}
      >
        <CardsWrapper expanded={expanded} ref={cardsRef}>
          {renderAssets()}
        </CardsWrapper>
      </Scrollbars>
    </>
  );
};

export default CardsHolder;
