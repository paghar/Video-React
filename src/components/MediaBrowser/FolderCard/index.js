//@flow
import React from 'react';
import FolderCardListView from 'components/MediaBrowser/FolderCard/FolderCardListView';
import { Display2, Display1 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import {
  FolderWrapper,
  Thumbnail,
  Numbers,
  More,
  FolderName,
  ImageWrapper,
  FileNumber,
  FolderInfo,
  FileCountWrapper,
  LastThumbnailWrapper,
  DropLayer,
  DropLayerAddIcon,
  DropLayerCaption,
  Wrapper,
} from './index.style';

export type FolderCardType = {
  selected?: boolean,
  thumbnails: Array<string>,
  fileCount: number,
  uploader: string,
  size?: string,
  uploadDate: number,
  name: string,
  expanded?: boolean,
};

type Props = {
  isDraggingOver?: boolean,
  dragging?: boolean,
  width: number,
  className?: string,
  clicked: (event: SyntheticMouseEvent<HTMLInputElement>) => void,
};

const FolderCard = ({
  thumbnails,
  uploadDate,
  expanded,
  fileCount,
  name,
  className,
  width,
  uploader,
  size,
  selected,
  clicked,
  isDraggingOver,
  dragging,
}: Props & FolderCardType) => {
  return (
    <Wrapper
      onMouseDown={clicked}
      width={width}
      expanded={expanded}
      selected={selected}
      className={className}
      isDraggingOver={isDraggingOver}
      dragging={dragging}
    >
      <DropLayer dragging={dragging} expanded={expanded}>
        <DropLayerAddIcon
          fontSize="35px"
          color={Generals.viii}
          iconNumber="61543"
        />
        <DropLayerCaption font={Display2} hoverable color={Generals.viii}>
          Add to Folder
        </DropLayerCaption>
      </DropLayer>
      <FolderWrapper
        isDraggingOver={isDraggingOver}
        expanded={expanded}
        width={width}
        fileCount={fileCount}
        selected={selected}
      >
        {!expanded && (
          <ImageWrapper>
            {thumbnails.slice(0, 4).map((thumbnail, id) => {
              return (
                <Thumbnail key={id} fileCount={fileCount} src={thumbnail} />
              );
            })}
            {fileCount >= 5 && (
              <LastThumbnailWrapper>
                {thumbnails.length > 4 && (
                  <Thumbnail
                    key={5}
                    fileCount={fileCount}
                    src={thumbnails[4]}
                  />
                )}
                {fileCount > 5 && (
                  <Numbers
                    hoverable
                    fileCount={fileCount}
                    font={Display1}
                    color={Generals.viii}
                  />
                )}
              </LastThumbnailWrapper>
            )}
          </ImageWrapper>
        )}

        {!expanded && (
          <FolderInfo>
            <FolderName hoverable font={Display2} color={Generals.viii}>
              {name}
            </FolderName>
            <FileCountWrapper>
              <FileNumber hoverable font={Display2} color={Generals.vii}>
                {fileCount} files
              </FileNumber>
              <More
                backgroundColor="transparent"
                hoverBackgroundColor="transparent"
                hoverColor={Generals.vii}
                radius="0px"
                iconSize="12px"
                icon="61762"
                spacing="0"
                width="auto"
                height="auto"
                iconColor={Generals.vii}
                hoverIconColor={Generals.vii}
              />
            </FileCountWrapper>
          </FolderInfo>
        )}
      </FolderWrapper>
      {expanded && (
        <FolderCardListView
          dragging={dragging}
          selected={selected}
          name={name}
          fileCount={fileCount}
          uploader={uploader}
          size={size}
          uploadDate={uploadDate}
        />
      )}
    </Wrapper>
  );
};

FolderCard.defaultProps = {
  name: '',
  fileCount: 0,
  thumbnails: [],
  uploader: '',
  uploadDate: 0,
  size: '',
};

export default FolderCard;
