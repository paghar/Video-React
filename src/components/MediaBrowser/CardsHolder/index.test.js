//@flow
import React from 'react';
import CardsHolder from '.';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import image from 'assets/image/Colleagues_looking_at_information_on_mobile_phone_screen_Videos_Mixkit.png';
import { FOLDER_TYPE, FILE_TYPE } from 'constants/variables';
import type { CardsType } from 'components/MediaBrowser/CardsHolder';
import FolderCard from 'components/MediaBrowser/FolderCard';

import {
  SELECT_ASSET,
  START_DRAG,
  ADD_ASSET_TO_SELECTION,
} from 'pages/MediaBrowser/index.actions';

const assets: CardsType = {
  '0': {
    type: FOLDER_TYPE,
    name: 'Hiking Experience',
    fileCount: 0,
    size: '16 MB',
    thumbnails: [],
    customOrder: 1,
    uploadDate: 1553277015000,
    uploader: 'Emily Parkson',
  },
  '1': {
    type: FOLDER_TYPE,
    name: 'Going into mountains Experience',
    fileCount: 0,
    size: '16 MB',
    thumbnails: [],
    customOrder: 2,
    uploadDate: 1553277015000,
    uploader: 'Emily Parkson',
  },
};

describe('<CardsHolder />', () => {
  it('checks a card on click', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <CardsHolder
        dropedOnAsset={console.log}
        dropedOnGutter={console.log}
        expanded={false}
        assetsList={assets}
        dispatchAssetsList={mockCallBack}
      />
    );
    wrapper
      .find(FolderCard)
      .at(0)
      .simulate('mousedown');
    expect(mockCallBack.mock.calls[0][0]).toEqual({
      assetId: '0',
      type: SELECT_ASSET,
    });
    wrapper
      .find(FolderCard)
      .at(1)
      .simulate('mousedown', { ctrlKey: true });

    expect(mockCallBack.mock.calls[2][0]).toEqual({
      assetId: '1',
      type: ADD_ASSET_TO_SELECTION,
    });
  });
  it('checks snapshot of holder', () => {
    const wrapper = renderer
      .create(
        <CardsHolder
          dropedOnAsset={console.log}
          dropedOnGutter={console.log}
          expanded={false}
          assetsList={assets}
          dispatchAssetsList={console.log}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
