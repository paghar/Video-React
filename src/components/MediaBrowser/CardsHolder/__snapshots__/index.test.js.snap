// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsHolder /> checks snapshot of holder 1`] = `
.c7 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  font-family: 'FontAwesome' !important;
  font-size: 35px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c19 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c20 {
  font-size: 12px;
  display: block;
  padding-right: 0;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c16 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: transparent;
  width: auto;
  height: auto;
  border-radius: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c16:hover {
  background-color: transparent;
}

.c6 {
  margin-top: -14px;
}

.c8 {
  text-align: center;
  margin-top: 7px;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 19;
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.28);
}

.c4::after {
  content: '';
  position: absolute;
  width: 23px;
  height: 9px;
  top: 0;
  left: 111px;
  background-color: #263c75;
}

.c18 {
  margin: 10px 2px 0 0;
  padding-right: 0px;
  display: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  cursor: pointer;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  min-width: 100%;
  padding-left: 9px;
  padding-right: 9px;
  padding-bottom: 14px;
  padding-top: 9px;
  margin-top: 0;
  border-radius: 0 7px 7px 7px;
  box-shadow: 0 0 0 3px transparent;
  z-index: 17;
}

.c10::after {
  content: '';
  display: inline-block;
  position: absolute;
  box-shadow: -3px -2px 0 1px transparent;
  z-index: -1;
  top: -8px;
  left: 111px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  width: 20px;
  height: 12px;
}

.c10::before {
  z-index: 0;
  box-shadow: -2px -3px 0 1px transparent;
  content: '';
  display: inline-block;
  position: absolute;
  top: -14px;
  left: 0px;
  width: 118px;
  height: 14px;
  border-radius: 7px 0 0 0;
}

.c10:hover .c17 {
  display: block;
}

.c11 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 138px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 0 0 0 2px;
  height: 20px;
}

.c13 {
  white-space: nowrap;
  margin: 8px 0 0 2px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: block;
  outline: none;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c15 {
  outline: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 8px 2px 0 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 250px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 15px;
  padding: 0;
  background-color: transparent;
  border-radius: 0px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 250ms;
  -webkit-transition: transform 250ms;
  transition: transform 250ms;
  pointer-events: auto;
}

.c2:hover {
  cursor: pointer;
}

.c2:hover .c25 {
  visibility: visible !important;
}

.c24:hover .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c24:hover .c9 {
  box-shadow: 0 0 0 3px;
}

.c24:hover .c9::after {
  background-color: #263c75;
  box-shadow: -3px -2px 0 1px;
}

.c24:hover .c9::before {
  background-color: #263c75;
  box-shadow: -2px -3px 0 1px;
}

.c23 {
  border-radius: 10px;
  width: 3px !important;
  opacity: 0.3;
}

.c22 {
  height: 100%;
  right: 0px;
  margin-top: 5px;
  border-radius: 10px;
  width: 2px !important;
  padding: 5px 0px;
}

.c21 {
  border: none;
  width: 21px;
  height: 190px;
  content: ' ';
}

.c21::after {
  content: '';
  position: absolute;
  margin-left: 11px;
  margin-top: 12px;
  width: 2px;
  display: none;
  height: 178px;
}

.c1 {
  border: none;
  width: 21px;
  height: 190px;
  content: ' ';
  position: absolute;
  z-index: 999;
}

.c1::after {
  content: '';
  position: absolute;
  margin-left: 11px;
  margin-top: 12px;
  width: 2px;
  display: none;
  height: 178px;
}

.c1::after {
  margin-left: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 98%;
  margin-left: 30px;
  margin-right: 0px;
  margin-bottom: 15px;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c0 > figure {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 20;
  width: auto;
  height: auto;
  margin-left: 3px;
  margin-bottom: 6px;
  margin-right: 0;
  margin-top: 3px;
}

.c0 > figure[data-type='folder'] {
  margin-bottom: 9px;
}

.c0 > figure[data-type='asset'] {
  margin-top: 6px;
}

@media (max-width:1140px) {
  .c2 {
    width: 250px;
  }
}

<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "bottom": 0,
        "left": 0,
        "marginBottom": 0,
        "marginRight": 0,
        "overflow": "scroll",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      className="c0"
    >
      <figure
        className="react-draggable"
        data-id="0"
        data-order={1}
        data-type="folder"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "order": 1,
            "touchAction": "none",
            "transform": "translate(0px,0px)",
          }
        }
        transform={null}
      >
        <hr
          className="c1"
          data-id="-1"
          data-type="gutter"
        />
        <div
          className="c2"
          onMouseDown={[Function]}
          width={250}
        >
          <div
            className="c3 c4"
          >
            <span
              className="c5 c6"
              fontSize="35px"
            >
              ÔÅß
            </span>
            <span
              className="c7 c8"
            >
              Add to Folder
            </span>
          </div>
          <div
            className="c9 c10"
            width={250}
          >
            <div
              className="c11"
            />
            <div
              className="c12"
            >
              <span
                className="c7 c13"
              >
                Hiking Experience
              </span>
              <div
                className="c14"
              >
                <span
                  className="c7 c15"
                >
                  0
                   files
                </span>
                <button
                  className="c16 c17 c18"
                  height="auto"
                  radius="0px"
                  width="auto"
                >
                  <span
                    className="c19 c20"
                    fontSize="12px"
                  >
                    ÔÖÇ
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <hr
          className="c21"
          data-id="0"
          data-type="gutter"
        />
      </figure>
      <figure
        className="react-draggable"
        data-id="1"
        data-order={2}
        data-type="folder"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "order": 2,
            "touchAction": "none",
            "transform": "translate(0px,0px)",
          }
        }
        transform={null}
      >
        <div
          className="c2"
          onMouseDown={[Function]}
          width={250}
        >
          <div
            className="c3 c4"
          >
            <span
              className="c5 c6"
              fontSize="35px"
            >
              ÔÅß
            </span>
            <span
              className="c7 c8"
            >
              Add to Folder
            </span>
          </div>
          <div
            className="c9 c10"
            width={250}
          >
            <div
              className="c11"
            />
            <div
              className="c12"
            >
              <span
                className="c7 c13"
              >
                Going into mountains Experience
              </span>
              <div
                className="c14"
              >
                <span
                  className="c7 c15"
                >
                  0
                   files
                </span>
                <button
                  className="c16 c17 c18"
                  height="auto"
                  radius="0px"
                  width="auto"
                >
                  <span
                    className="c19 c20"
                    fontSize="12px"
                  >
                    ÔÖÇ
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <hr
          className="c21"
          data-id="1"
          data-type="gutter"
        />
      </figure>
    </div>
  </div>
  <div
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "height": 6,
        "left": 2,
        "opacity": 0,
        "position": "absolute",
        "right": 2,
        "transition": "opacity 200ms",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.2)",
          "borderRadius": "inherit",
          "cursor": "pointer",
          "display": "block",
          "height": "100%",
          "position": "relative",
        }
      }
    />
  </div>
  <div
    className="c22"
    style={
      Object {
        "display": "none",
        "opacity": 0,
        "position": "absolute",
        "transition": "opacity 200ms",
        "width": 6,
      }
    }
  >
    <div
      className="c23"
      style={
        Object {
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
